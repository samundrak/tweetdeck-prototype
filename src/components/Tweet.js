import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Skeleton, Card, Icon, Avatar } from 'antd';
import styled from 'styled-components';
import distanceInWordsToNow from 'date-fns/distance_in_words_to_now';
import TwitterLink from './TwitterLink';

const { Meta } = Card;
const CreatedAt = styled.div`
  position: absolute;
  float: right;
  right: 0px;
  padding-right: 10px;
  margin-top: 0px;
  color: darkgray;
`;
const Counts = styled.span``;

const gotoTweet = (handle, tweetId) => () => window.open(`https://twitter.com/${handle}/status/${tweetId}`);
const Tweet = ({ tweet, handleActions }) => (
  <Card
    style={{}}
    actions={[
      <Fragment>
        <div tabIndex="0" role="link" onClick={handleActions({ type: 'retweets', tweet })}>
          <Icon type="retweet" title="Retweets" />
          <Counts> {tweet.retweet_count}</Counts>
        </div>
      </Fragment>,
      <Fragment>
        <div tabIndex="0" role="link" onClick={handleActions({ type: 'mentions', tweet })}>
          <Icon type="tag" title="mentions" />
          <Counts> {tweet.entities.user_mentions.length || ''}</Counts>
        </div>
      </Fragment>,
    ]}
  >
    <Skeleton loading={false} avatar active>
      <CreatedAt>{distanceInWordsToNow(tweet.created_at)} ago</CreatedAt>
      <Meta
        avatar={<Avatar src={tweet.user.profile_image_url} />}
        title={<TwitterLink handle={tweet.user.screen_name} />}
        description={
          <div tabIndex="0" role="link" style={{ cursor: 'pointer' }} onClick={gotoTweet(tweet.user.screen_name, tweet.id_str)}>
            {tweet.text}
          </div>
        }
      />
    </Skeleton>
  </Card>
);
Tweet.propTypes = {
  tweet: PropTypes.object.isRequired,
  handleActions: PropTypes.func.isRequired,
};
export default Tweet;
