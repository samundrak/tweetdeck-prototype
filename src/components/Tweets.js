import React from 'react';
import Tweet from './Tweet';

const Tweets = ({ tweets }) => {
  return (
    <div>
      {tweets.map(tweet => (
        <Tweet tweet={tweet} key={tweet.id} />
      ))}
    </div>
  );
};
export default Tweets;
