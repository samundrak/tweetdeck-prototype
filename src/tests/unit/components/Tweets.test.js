import React from 'react';
import { mount } from 'enzyme';
import tweets from '../../tweets.json';
import Tweets from '../../../components/Tweets';
import Tweet from '../../../components/Tweet';

describe('Test Tweets', () => {
  it('should render list of tweets', () => {
    const mockFn = () => null;
    const wrapper = mount(<Tweets tweets={tweets} handleActions={mockFn} />);
    expect(wrapper.find(Tweet).length).toBe(4);
  });
  it('should test snapshot', () => {
    const mockFn = () => null;
    const wrapper = mount(<Tweets tweets={tweets} handleActions={mockFn} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
