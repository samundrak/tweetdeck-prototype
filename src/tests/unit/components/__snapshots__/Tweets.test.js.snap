// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tweets should test snapshot 1`] = `"<div><div class=\\"ant-card ant-card-bordered\\"><div class=\\"ant-card-body\\"><div class=\\"sc-bdVaJa jLiEut\\">about 2 hours ago</div><div class=\\"ant-card-meta\\"><div class=\\"ant-card-meta-avatar\\"><span class=\\"ant-avatar ant-avatar-circle ant-avatar-image\\"><img src=\\"http://pbs.twimg.com/profile_images/469397708986269696/iUrYEOpJ_normal.png\\"></span></div><div class=\\"ant-card-meta-detail\\"><div class=\\"ant-card-meta-title\\"><a href=\\"https://twitter.com/newsycombinator\\" target=\\"_blank\\"><div></div></a></div><div class=\\"ant-card-meta-description\\"><div tabindex=\\"0\\" role=\\"link\\" style=\\"cursor: pointer;\\">Migraines from LED light/PWM https://t.co/qlrlvR1rzR</div></div></div></div></div><ul class=\\"ant-card-actions\\"><li style=\\"width: 50%;\\"><span><div tabindex=\\"0\\" role=\\"link\\"><i title=\\"Retweets\\" class=\\"anticon anticon-retweet\\"><svg viewBox=\\"0 0 1024 1024\\" class=\\"\\" data-icon=\\"retweet\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M136 552h63.6c4.4 0 8-3.6 8-8V288.7h528.6v72.6c0 1.9.6 3.7 1.8 5.2a8.3 8.3 0 0 0 11.7 1.4L893 255.4c4.3-5 3.6-10.3 0-13.2L749.7 129.8a8.22 8.22 0 0 0-5.2-1.8c-4.6 0-8.4 3.8-8.4 8.4V209H199.7c-39.5 0-71.7 32.2-71.7 71.8V544c0 4.4 3.6 8 8 8zm752-80h-63.6c-4.4 0-8 3.6-8 8v255.3H287.8v-72.6c0-1.9-.6-3.7-1.8-5.2a8.3 8.3 0 0 0-11.7-1.4L131 768.6c-4.3 5-3.6 10.3 0 13.2l143.3 112.4c1.5 1.2 3.3 1.8 5.2 1.8 4.6 0 8.4-3.8 8.4-8.4V815h536.6c39.5 0 71.7-32.2 71.7-71.8V480c-.2-4.4-3.8-8-8.2-8z\\"></path></svg></i><span class=\\"sc-bwzfXH uTxCW\\"> 1</span></div></span></li><li style=\\"width: 50%;\\"><span><div tabindex=\\"0\\" role=\\"link\\"><i title=\\"mentions\\" class=\\"anticon anticon-tag\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"tag\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M938 458.8l-29.6-312.6c-1.5-16.2-14.4-29-30.6-30.6L565.2 86h-.4c-3.2 0-5.7 1-7.6 2.9L88.9 557.2a9.96 9.96 0 0 0 0 14.1l363.8 363.8c1.9 1.9 4.4 2.9 7.1 2.9s5.2-1 7.1-2.9l468.3-468.3c2-2.1 3-5 2.8-8zM459.7 834.7L189.3 564.3 589 164.6 836 188l23.4 247-399.7 399.7zM680 256c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm0 120c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z\\"></path></svg></i><span class=\\"sc-bwzfXH uTxCW\\"> </span></div></span></li></ul></div><div class=\\"ant-card ant-card-bordered\\"><div class=\\"ant-card-body\\"><div class=\\"sc-bdVaJa jLiEut\\">about 2 hours ago</div><div class=\\"ant-card-meta\\"><div class=\\"ant-card-meta-avatar\\"><span class=\\"ant-avatar ant-avatar-circle ant-avatar-image\\"><img src=\\"http://pbs.twimg.com/profile_images/469397708986269696/iUrYEOpJ_normal.png\\"></span></div><div class=\\"ant-card-meta-detail\\"><div class=\\"ant-card-meta-title\\"><a href=\\"https://twitter.com/newsycombinator\\" target=\\"_blank\\"><div></div></a></div><div class=\\"ant-card-meta-description\\"><div tabindex=\\"0\\" role=\\"link\\" style=\\"cursor: pointer;\\">Show HN: A mobile friendly web based countries of the world game https://t.co/pl6eLNkEYD</div></div></div></div></div><ul class=\\"ant-card-actions\\"><li style=\\"width: 50%;\\"><span><div tabindex=\\"0\\" role=\\"link\\"><i title=\\"Retweets\\" class=\\"anticon anticon-retweet\\"><svg viewBox=\\"0 0 1024 1024\\" class=\\"\\" data-icon=\\"retweet\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M136 552h63.6c4.4 0 8-3.6 8-8V288.7h528.6v72.6c0 1.9.6 3.7 1.8 5.2a8.3 8.3 0 0 0 11.7 1.4L893 255.4c4.3-5 3.6-10.3 0-13.2L749.7 129.8a8.22 8.22 0 0 0-5.2-1.8c-4.6 0-8.4 3.8-8.4 8.4V209H199.7c-39.5 0-71.7 32.2-71.7 71.8V544c0 4.4 3.6 8 8 8zm752-80h-63.6c-4.4 0-8 3.6-8 8v255.3H287.8v-72.6c0-1.9-.6-3.7-1.8-5.2a8.3 8.3 0 0 0-11.7-1.4L131 768.6c-4.3 5-3.6 10.3 0 13.2l143.3 112.4c1.5 1.2 3.3 1.8 5.2 1.8 4.6 0 8.4-3.8 8.4-8.4V815h536.6c39.5 0 71.7-32.2 71.7-71.8V480c-.2-4.4-3.8-8-8.2-8z\\"></path></svg></i><span class=\\"sc-bwzfXH uTxCW\\"> 1</span></div></span></li><li style=\\"width: 50%;\\"><span><div tabindex=\\"0\\" role=\\"link\\"><i title=\\"mentions\\" class=\\"anticon anticon-tag\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"tag\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M938 458.8l-29.6-312.6c-1.5-16.2-14.4-29-30.6-30.6L565.2 86h-.4c-3.2 0-5.7 1-7.6 2.9L88.9 557.2a9.96 9.96 0 0 0 0 14.1l363.8 363.8c1.9 1.9 4.4 2.9 7.1 2.9s5.2-1 7.1-2.9l468.3-468.3c2-2.1 3-5 2.8-8zM459.7 834.7L189.3 564.3 589 164.6 836 188l23.4 247-399.7 399.7zM680 256c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm0 120c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z\\"></path></svg></i><span class=\\"sc-bwzfXH uTxCW\\"> </span></div></span></li></ul></div><div class=\\"ant-card ant-card-bordered\\"><div class=\\"ant-card-body\\"><div class=\\"sc-bdVaJa jLiEut\\">about 2 hours ago</div><div class=\\"ant-card-meta\\"><div class=\\"ant-card-meta-avatar\\"><span class=\\"ant-avatar ant-avatar-circle ant-avatar-image\\"><img src=\\"http://pbs.twimg.com/profile_images/469397708986269696/iUrYEOpJ_normal.png\\"></span></div><div class=\\"ant-card-meta-detail\\"><div class=\\"ant-card-meta-title\\"><a href=\\"https://twitter.com/newsycombinator\\" target=\\"_blank\\"><div></div></a></div><div class=\\"ant-card-meta-description\\"><div tabindex=\\"0\\" role=\\"link\\" style=\\"cursor: pointer;\\">BÃ©zier moi https://t.co/kLyHxePWvP</div></div></div></div></div><ul class=\\"ant-card-actions\\"><li style=\\"width: 50%;\\"><span><div tabindex=\\"0\\" role=\\"link\\"><i title=\\"Retweets\\" class=\\"anticon anticon-retweet\\"><svg viewBox=\\"0 0 1024 1024\\" class=\\"\\" data-icon=\\"retweet\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M136 552h63.6c4.4 0 8-3.6 8-8V288.7h528.6v72.6c0 1.9.6 3.7 1.8 5.2a8.3 8.3 0 0 0 11.7 1.4L893 255.4c4.3-5 3.6-10.3 0-13.2L749.7 129.8a8.22 8.22 0 0 0-5.2-1.8c-4.6 0-8.4 3.8-8.4 8.4V209H199.7c-39.5 0-71.7 32.2-71.7 71.8V544c0 4.4 3.6 8 8 8zm752-80h-63.6c-4.4 0-8 3.6-8 8v255.3H287.8v-72.6c0-1.9-.6-3.7-1.8-5.2a8.3 8.3 0 0 0-11.7-1.4L131 768.6c-4.3 5-3.6 10.3 0 13.2l143.3 112.4c1.5 1.2 3.3 1.8 5.2 1.8 4.6 0 8.4-3.8 8.4-8.4V815h536.6c39.5 0 71.7-32.2 71.7-71.8V480c-.2-4.4-3.8-8-8.2-8z\\"></path></svg></i><span class=\\"sc-bwzfXH uTxCW\\"> 2</span></div></span></li><li style=\\"width: 50%;\\"><span><div tabindex=\\"0\\" role=\\"link\\"><i title=\\"mentions\\" class=\\"anticon anticon-tag\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"tag\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M938 458.8l-29.6-312.6c-1.5-16.2-14.4-29-30.6-30.6L565.2 86h-.4c-3.2 0-5.7 1-7.6 2.9L88.9 557.2a9.96 9.96 0 0 0 0 14.1l363.8 363.8c1.9 1.9 4.4 2.9 7.1 2.9s5.2-1 7.1-2.9l468.3-468.3c2-2.1 3-5 2.8-8zM459.7 834.7L189.3 564.3 589 164.6 836 188l23.4 247-399.7 399.7zM680 256c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm0 120c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z\\"></path></svg></i><span class=\\"sc-bwzfXH uTxCW\\"> </span></div></span></li></ul></div><div class=\\"ant-card ant-card-bordered\\"><div class=\\"ant-card-body\\"><div class=\\"sc-bdVaJa jLiEut\\">about 6 hours ago</div><div class=\\"ant-card-meta\\"><div class=\\"ant-card-meta-avatar\\"><span class=\\"ant-avatar ant-avatar-circle ant-avatar-image\\"><img src=\\"http://pbs.twimg.com/profile_images/469397708986269696/iUrYEOpJ_normal.png\\"></span></div><div class=\\"ant-card-meta-detail\\"><div class=\\"ant-card-meta-title\\"><a href=\\"https://twitter.com/newsycombinator\\" target=\\"_blank\\"><div></div></a></div><div class=\\"ant-card-meta-description\\"><div tabindex=\\"0\\" role=\\"link\\" style=\\"cursor: pointer;\\">The Life of a Backpacker in Asia in the 1970s https://t.co/0WiTZeqgHV</div></div></div></div></div><ul class=\\"ant-card-actions\\"><li style=\\"width: 50%;\\"><span><div tabindex=\\"0\\" role=\\"link\\"><i title=\\"Retweets\\" class=\\"anticon anticon-retweet\\"><svg viewBox=\\"0 0 1024 1024\\" class=\\"\\" data-icon=\\"retweet\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M136 552h63.6c4.4 0 8-3.6 8-8V288.7h528.6v72.6c0 1.9.6 3.7 1.8 5.2a8.3 8.3 0 0 0 11.7 1.4L893 255.4c4.3-5 3.6-10.3 0-13.2L749.7 129.8a8.22 8.22 0 0 0-5.2-1.8c-4.6 0-8.4 3.8-8.4 8.4V209H199.7c-39.5 0-71.7 32.2-71.7 71.8V544c0 4.4 3.6 8 8 8zm752-80h-63.6c-4.4 0-8 3.6-8 8v255.3H287.8v-72.6c0-1.9-.6-3.7-1.8-5.2a8.3 8.3 0 0 0-11.7-1.4L131 768.6c-4.3 5-3.6 10.3 0 13.2l143.3 112.4c1.5 1.2 3.3 1.8 5.2 1.8 4.6 0 8.4-3.8 8.4-8.4V815h536.6c39.5 0 71.7-32.2 71.7-71.8V480c-.2-4.4-3.8-8-8.2-8z\\"></path></svg></i><span class=\\"sc-bwzfXH uTxCW\\"> 2</span></div></span></li><li style=\\"width: 50%;\\"><span><div tabindex=\\"0\\" role=\\"link\\"><i title=\\"mentions\\" class=\\"anticon anticon-tag\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"tag\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M938 458.8l-29.6-312.6c-1.5-16.2-14.4-29-30.6-30.6L565.2 86h-.4c-3.2 0-5.7 1-7.6 2.9L88.9 557.2a9.96 9.96 0 0 0 0 14.1l363.8 363.8c1.9 1.9 4.4 2.9 7.1 2.9s5.2-1 7.1-2.9l468.3-468.3c2-2.1 3-5 2.8-8zM459.7 834.7L189.3 564.3 589 164.6 836 188l23.4 247-399.7 399.7zM680 256c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm0 120c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z\\"></path></svg></i><span class=\\"sc-bwzfXH uTxCW\\"> </span></div></span></li></ul></div></div>"`;
